### DISCUSSION TITLE
Game Mode

### DISCUSSION THREAD

**Discussion by ghost** [saul/demofile#263] [Questions] [answered]
Hi, it's possible to get the Game Mode ? Like "Competitve" or "Wingman" or "Scrimmage" etc ?

**Comment by saul** [comment, marked as answer]
You can get the game type and game mode with:

```ts
var gameType = demoFile.conVars.vars.get("game_type") | 0;
var gameMode = demoFile.conVars.vars.get("game_mode") | 0;
```

| \#  | Game type    | Game modes                                                                |
| --- | ------------ | ------------------------------------------------------------------------- |
| 0   | Classic      | 0 = Casual<br>1 = Competitive<br>2 = Scrim comp 2v2<br>3 = Scrim comp 5v5 |
| 1   | GunGame      | 0 = Progressive<br>1 = Bomb<br>2 = Deathmatch                             |
| 2   | Training     | 0 = Training                                                              |
| 3   | Custom       | 0 = Custom                                                                |
| 4   | Co-operative | 0 = Co-operative<br>1 = Co-op mission                                     |
| 5   | Skirmish     | 0 = Skirmish                                                              |
| 6   | Free for all | 0 = Survival                                                              |

So `game_type = 1` and `game_mode = 2` corresponds to GunGame Deathmatch.

You can find the full information here: https://github.com/SteamDatabase/GameTracking-CSGO/blob/master/csgo/gamemodes.txt

Hope that makes sense.
